katharsis-core/src/main/java/io/katharsis/legacy/queryParams/DefaultQueryParamsConverter.java

```diff
@@ -0,0 +1,138 @@
+package io.katharsis.legacy.queryParams;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+import io.katharsis.core.internal.utils.StringUtils;
+import io.katharsis.legacy.queryParams.include.Inclusion;
+import io.katharsis.legacy.queryParams.params.*;
+import io.katharsis.queryspec.*;
+import io.katharsis.resource.information.ResourceInformation;
+import io.katharsis.resource.registry.RegistryEntry;
+import io.katharsis.resource.registry.ResourceRegistry;
+
+@SuppressWarnings({"deprecation"})
+public class DefaultQueryParamsConverter implements QueryParamsConverter {
+
+    private ResourceRegistry resourceRegistry;
+
+    public DefaultQueryParamsConverter(ResourceRegistry resourceRegistry) {
+        this.resourceRegistry = resourceRegistry;
+    }
+
+    @Override
+    public QueryParams fromParams(Class<?> rootType, QuerySpec querySpec) {
+        QueryParams params = new QueryParams();
```

Comment:
you may want to call getRelatedSpecs as well and serialize those QuerySpecs as well (to process parameters other than for the root type)